### ISSUE TITLE
LocationUpdateTask global workqueue cherrypy thread crashing with timeout

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#9668]
**Impact of the bug**
Global Workqueue

**Describe the bug**
With the avalanche of workflows in the system (3 NanoAOD campaigns injected in parallel), we have stressed the central services to an unprecedented load, resulting in 45k active workflows in the system (22k being in `acquired` status) and spawning 313k GQEs in `Available` status(!!!).

After this introduction, global workqueue went into red state in wmstats for the last ~7h, with two of its cherrypy threads. Looking into the locationUpdateTask logs in vocms055, I see that the thread is actually timing out during the location update task. Traceback will be provided below.

This is quite important to get fixed, likely not critical though: GQEs will not have their location updated every X hours. Instead, they will contain the initial data location when the GQE got created.

**How to reproduce it**
Creating hundreds of thousands of GQEs(?)

**Expected behavior**
The obvious one, LocationUpdateTask cherrypy thread should not crash.
We need to revisit this logic and make sure that GQEs are fetched/updated in slices, such that we can avoid fetching a lot of data in one call (and avoid possible timeouts).

**Additional context and error message**
Whole traceback from locationUpdateTask-vocms0740-20200429.log
```
2020-04-29 12:41:34,108:INFO:WorkQueue:Executing data location update...
2020-04-29 12:46:04,155:ERROR:CherryPyPeriodicTask:Periodic Thread ERROR pycurl.error (28, 'Operation timed out after 270000 milliseconds with 31852063 bytes received')
2020-04-29 12:46:04,156:ERROR:CherryPyPeriodicTask: Traceback (most recent call last):
2020-04-29 12:46:04,156:ERROR:CherryPyPeriodicTask:   File "/data/srv/beHG2004f/sw/slc7_amd64_gcc630/cms/workqueue/1.3.2.pre5/lib/python2.7/site-packages/WMCore/REST/CherryPyPeriodicTask.py", line 91, in run
2020-04-29 12:46:04,156:ERROR:CherryPyPeriodicTask:     self.taskFunc(self.config)
2020-04-29 12:46:04,156:ERROR:CherryPyPeriodicTask:   File "/data/srv/beHG2004f/sw/slc7_amd64_gcc630/cms/workqueue/1.3.2.pre5/lib/python2.7/site-packages/WMCore/GlobalWorkQueue/CherryPyThreads/LocationUpdateTask.py", line 24, in updateDataLocation
2020-04-29 12:46:04,156:ERROR:CherryPyPeriodicTask:     res = globalQ.updateLocationInfo()
2020-04-29 12:46:04,156:ERROR:CherryPyPeriodicTask:   File "/data/srv/beHG2004f/sw/slc7_amd64_gcc630/cms/workqueue/1.3.2.pre5/lib/python2.7/site-packages/WMCore/WorkQueue/WorkQueue.py", line 730, in updateLocationInfo
2020-04-29 12:46:04,156:ERROR:CherryPyPeriodicTask:     result = self.dataLocationMapper()
2020-04-29 12:46:04,156:ERROR:CherryPyPeriodicTask:   File "/data/srv/beHG2004f/sw/slc7_amd64_gcc630/cms/workqueue/1.3.2.pre5/lib/python2.7/site-packages/WMCore/WorkQueue/DataLocationMapper.py", line 184, in __call__
2020-04-29 12:46:04,157:ERROR:CherryPyPeriodicTask:     dataItems = self.backend.getActiveData()
2020-04-29 12:46:04,157:ERROR:CherryPyPeriodicTask:   File "/data/srv/beHG2004f/sw/slc7_amd64_gcc630/cms/workqueue/1.3.2.pre5/lib/python2.7/site-packages/WMCore/WorkQueue/WorkQueueBackend.py", line 424, in getActiveData
2020-04-29 12:46:04,157:ERROR:CherryPyPeriodicTask:     data = self.db.loadView('WorkQueue', 'activeData', {'reduce': True, 'group': True})
2020-04-29 12:46:04,157:ERROR:CherryPyPeriodicTask:   File "/data/srv/beHG2004f/sw/slc7_amd64_gcc630/cms/workqueue/1.3.2.pre5/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 537, in loadView
2020-04-29 12:46:04,157:ERROR:CherryPyPeriodicTask:     (self.name, design, view), encodedOptions)
2020-04-29 12:46:04,157:ERROR:CherryPyPeriodicTask:   File "/data/srv/beHG2004f/sw/slc7_amd64_gcc630/cms/workqueue/1.3.2.pre5/lib/python2.7/site-packages/WMCore/Services/Requests.py", line 127, in get
2020-04-29 12:46:04,157:ERROR:CherryPyPeriodicTask:     encode, decode, contentType)
2020-04-29 12:46:04,157:ERROR:CherryPyPeriodicTask:   File "/data/srv/beHG2004f/sw/slc7_amd64_gcc630/cms/workqueue/1.3.2.pre5/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 122, in makeRequest
2020-04-29 12:46:04,158:ERROR:CherryPyPeriodicTask:     encode, decode, contentType)
2020-04-29 12:46:04,158:ERROR:CherryPyPeriodicTask:   File "/data/srv/beHG2004f/sw/slc7_amd64_gcc630/cms/workqueue/1.3.2.pre5/lib/python2.7/site-packages/WMCore/Services/Requests.py", line 165, in makeRequest
2020-04-29 12:46:04,158:ERROR:CherryPyPeriodicTask:     result, response = self.makeRequest_pycurl(uri, data, verb, headers)
2020-04-29 12:46:04,158:ERROR:CherryPyPeriodicTask:   File "/data/srv/beHG2004f/sw/slc7_amd64_gcc630/cms/workqueue/1.3.2.pre5/lib/python2.7/site-packages/WMCore/Services/Requests.py", line 183, in makeRequest_pycurl
2020-04-29 12:46:04,158:ERROR:CherryPyPeriodicTask:     ckey=ckey, cert=cert, capath=capath)
2020-04-29 12:46:04,158:ERROR:CherryPyPeriodicTask:   File "/data/srv/beHG2004f/sw/slc7_amd64_gcc630/cms/workqueue/1.3.2.pre5/lib/python2.7/site-packages/WMCore/Services/pycurl_manager.py", line 257, in request
2020-04-29 12:46:04,158:ERROR:CherryPyPeriodicTask:     curl.perform()
2020-04-29 12:46:04,158:ERROR:CherryPyPeriodicTask: error: (28, 'Operation timed out after 270000 milliseconds with 31852063 bytes received')
```

**Comment by amaltaro** [comment]
Duplicate of https://github.com/dmwm/WMCore/issues/12319, which has been recently fixed and deployed in production with `2.3.8.5`.
